codesign/codesign.go

```diff
@@ -250,22 +260,45 @@ func (m *Manager) selectCodeSigningStrategy(credentials appleauth.Credentials) (
 	return codeSigningXcode, "Automatically managed signing is enabled in Xcode for the project.", nil
 }
 
-func (m *Manager) downloadAndInstallCertificates() error {
+func (m *Manager) downloadCertificates() ([]certificateutil.CertificateInfoModel, error) {
 	certificates, err := m.certDownloader.GetCertificates()
 	if err != nil {
-		return fmt.Errorf("failed to download certificates: %s", err)
+		return nil, fmt.Errorf("failed to download certificates: %s", err)
 	}
 
-	certificateType, ok := autocodesign.CertificateTypeByDistribution[m.opts.ExportMethod]
-	if !ok {
-		panic(fmt.Sprintf("no valid certificate provided for distribution type: %s", m.opts.ExportMethod))
+	if len(certificates) == 0 {
+		m.logger.Warnf("No certificates are uploaded to Bitrise.")
+
+		return nil, nil
 	}
 
+	return certificates, nil
+}
+
+func (m *Manager) installCertificates(certificates []certificateutil.CertificateInfoModel) error {
+	m.logger.Infof("Installing certificates:")
+	for _, cert := range certificates {
+		m.logger.Printf("- %s", cert)
+		// Empty passphrase provided, as already parsed certificate + private key
+		if err := m.assetInstaller.InstallCertificate(cert); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (m *Manager) checkXcodeManagedCertificates(certificates []certificateutil.CertificateInfoModel) error {
```

Comment:
I do not see how the function name relates to the function body. I mean I do not get why the XcodeManaged part.